source/common/network/connection_impl.cc

```diff
@@ -247,6 +247,8 @@ void ConnectionImpl::closeSocket(ConnectionEvent close_type) {
     delayed_close_timer_ = nullptr;
   }
 
+  stream_info_.setDownstreamTransportFailureReason(transport_socket_->failureReason());
```

Comment:
Can we separate this change? I'm not sure this is the right place to record the transport failure reason. It might be better to move it closer to where it logs()?

Reply:
/wait